src/ray/common/ray_config_def.h

```diff
@@ -818,6 +818,10 @@ RAY_CONFIG(bool, emit_event_to_log_file, false)
 /// Event severity threshold value
 RAY_CONFIG(std::string, event_level, "warning")
 
+/// RPC timeout for the metrics exporter to report metrics to the metrics agent.
+/// The default is 10 seconds. If set to -1, the RPC deadline will not be set.
+RAY_CONFIG(int64_t, metrics_report_rpc_timeout_ms, 10000)
```

Comment:
10 seconds is really really long, I use to set timeout 500 millisecond for request from us-west to us-east.

Timeout could be used to deal with straggler, a over-large timeout doesn't help.

Reply:
I usually set to a large value for safety (but not 10 sec vs 1 ms) and add metrics, so we could observe the effect and tune later based on metrics. If we know the average latency is 1 ms, 10 ms seems ok.

Reply:
After some investigation, I decided to set the default value to 500 ms. You can check the PR description for more details. Would you mind checking whether the reason makes sense to you or not?